server/controllers/wishlistController.js

```diff
@@ -0,0 +1,202 @@
+const Wishlist = require('../models/wishlist');
+
+const _getWishlistCost = list => {
+  // Accept list as Array of objects
+  // Should return a single cost for the entire wishlist
+  let totalCost = 0;
+
+  list.forEach(item => {
+    totalCost += item.itemCost * item.itemQuantity;
+  });
+
+  return totalCost;
+};
+
+exports.addItem = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const item = req.body.item;
+
+  // TODO: Check structure of item
+  // Prevent duplicate items
+
+  try {
+    Wishlist.find({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      wishlist[0].list.push(item);
+
+      wishlist[0].totalListCost = _getWishlistCost(wishlist[0].list);
+
+      wishlist[0].save((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        res.status(201).json({
+          message: 'Successfully added item to wishlist',
+          wishlist: wishlist,
+        });
+      });
+    });
+  } catch (err) {
+    res.status(500).json({ Message: 'TEMP ERROR TEST' });
+  }
+};
+
+exports.create = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const newWishlist = req.body.list;
+
+  // Check newWishlist object type is an Array
+  // Single items should still reside in an Array
+
+  try {
+    Wishlist.where({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      if (wishlist[0].list !== []) {
+        return res.status(200).json({
```

Comment:
Doesn't make sense for this to be a 200 response, possibly use a 409 or 412 response